src/crimson/os/seastore/cache.h

```diff
@@ -978,7 +978,8 @@ class Cache {
     auto result = epm.alloc_new_non_data_extent(t, T::TYPE, length, hint, gen);
 #endif
     if (!result) {
-      return nullptr;
```

Comment:
Not sure which issue is related -- why returning `nullptr` then convert it to `return crimson::ct_error::enospc::make();` has to be changed?

Reply:
Previously, only `TransactionManager` return `enospc` when hitting nullptr from `Cache::alloc_new_xxx_extent()`, which is insufficient. FixedKVBtree also need to return enospc in this case, but FixedKVBtree calls `Cache::alloc_new_xxx_extent()` in non-future-returning methods, which means it can't return `crimson::ct_error::enospc::make()`. So, instead of throw `enospc` in multiple places in FixedKVBtree, I throw it in `Cache`.